pkg/api/validate/dynamic/dynamic.go

```diff
@@ -458,9 +461,19 @@ func (c closure) usingCheckAccessV2() (bool, error) {
 		return false, nil
 	}
 
-	for _, result := range results.Value {
-		if result.AccessDecision != remotepdp.Allowed {
-			c.dv.log.Infof("%s has no access to %s", *c.oid, result.ActionId)
+	for _, action := range c.actions {
+		found := false
+		for _, result := range results.Value {
+			if result.ActionId == action {
+				found = true
+				if result.AccessDecision == remotepdp.Allowed {
+					break
+				}
+				return false, nil
```

Comment:
is there value in keeping this log here: `c.dv.log.Infof("%s has no access to %s", *c.oid, result.ActionId)`

Reply:
I'm springing a lot of logging statements here because the API hasn't been GAed yet and there are chances that it might give wrong results.  I plan to remove most of the loggings once we have completely switched over to the new API.  For now, I think the more data to observe how the API behaves, the better.

For the looping question, if we don't find a matching action, then it means the particular action/permission hasn't been granted to the service principal.  In that case, it should stop and go into another retry call.